From 0a95fa744b5bd387bcc148cd8c19fee5359d2110 Mon Sep 17 00:00:00 2001
From: Mohsen Dolaty <mohsen.dolaty@amd.com>
Date: Mon, 25 Jul 2022 13:10:28 -0500
Subject: [PATCH 1/1] linux-aspeed: Add DIMM devices for CH2 in Ruby

Ruby is a 2 DIMM per Ch system
Add DIMM devices for Ch 2

Signed-off-by: Mohsen Dolaty <mohsen.dolaty@amd.com>
---
 arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts | 173 ++++++++++++++++++++++
 1 file changed, 173 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
index ca7199d829f3..0767a1f36d96 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
@@ -619,6 +619,179 @@ rcd_1_5: rcd@5d,2C000000005 {
 		assigned-address = <0x5d>;
 	};
 };
+&i3c2 {
+        // P1 DIMM (A-F) SPD Access
+        status = "okay";
+
+        jdec-spd;
+        bus_id = <0x02>;
+
+        //DIMM A
+        spd5118_2_0: spd@50,3C000000000 {
+                reg = <0x50 0x3C0 0x00000000>;
+                assigned-address = <0x50>;
+        };
+        pmic5xxx_2_0: pmic@48,20400000000 {
+                reg = <0x48 0x204 0x00000000>;
+                assigned-address = <0x48>;
+        };
+        rcd_2_0: rcd@58,2C000000000 {
+                reg = <0x58 0x2C0 0x00000000>;
+                assigned-address = <0x58>;
+        };
+        //DIMM B
+        spd5118_2_1: spd@51,3C000000001 {
+                reg = <0x51 0x3C0 0x00000001>;
+                assigned-address = <0x51>;
+        };
+        pmic5xxx_2_1: pmic@49,20400000001 {
+                reg = <0x49 0x204 0x00000001>;
+                assigned-address = <0x49>;
+        };
+        rcd_2_1: rcd@59,2C000000001 {
+                reg = <0x59 0x2C0 0x00000001>;
+                assigned-address = <0x59>;
+        };
+        //DIMM C
+        spd5118_2_2: spd@52,3C000000002 {
+                reg = <0x52 0x3C0 0x00000002>;
+                assigned-address = <0x52>;
+        };
+        pmic5xxx_2_2: pmic@4a,20400000002 {
+                reg = <0x4a 0x204 0x00000002>;
+                assigned-address = <0x4a>;
+        };
+        rcd_2_2: rcd@5a,2C000000002 {
+                reg = <0x5a 0x2C0 0x00000002>;
+                assigned-address = <0x5a>;
+        };
+        //DIMM D
+        spd5118_2_3: spd@53,3C000000003 {
+                reg = <0x53 0x3C0 0x00000003>;
+                assigned-address = <0x53>;
+        };
+        pmic5xxx_2_3: pmic@4b,20400000003 {
+                reg = <0x4b 0x204 0x00000003>;
+                assigned-address = <0x4b>;
+        };
+        rcd_2_3: rcd@5b,2C000000003 {
+                reg = <0x5b 0x2C0 0x00000003>;
+                assigned-address = <0x5b>;
+        };
+        //DIMM E
+        spd5118_2_4: spd@54,3C000000004 {
+                reg = <0x54 0x3C0 0x00000004>;
+                assigned-address = <0x54>;
+        };
+        pmic5xxx_2_4: pmic@4c,20400000004 {
+                reg = <0x4c 0x204 0x00000004>;
+                assigned-address = <0x4c>;
+        };
+        rcd_2_4: rcd@5c,2C000000004 {
+                reg = <0x5c 0x2C0 0x00000004>;
+                assigned-address = <0x5c>;
+        };
+        //DIMM F
+        spd5118_2_5: spd@55,3C000000005 {
+                reg = <0x55 0x3C0 0x00000005>;
+                assigned-address = <0x55>;
+        };
+        pmic5xxx_2_5: pmic@4d,20400000005 {
+                reg = <0x4d 0x204 0x00000005>;
+                assigned-address = <0x4d>;
+        };
+        rcd_2_5: rcd@5d,2C000000005 {
+                reg = <0x5d 0x2C0 0x00000005>;
+                assigned-address = <0x5d>;
+        };
+};
+
+&i3c3 {
+        // P1 DIMM (G-L) SPD Access
+        status = "okay";
+
+        jdec-spd;
+        bus_id = <0x03>;
+
+        //DIMM G
+        spd5118_3_0: spd@50,3C000000000 {
+                reg = <0x50 0x3C0 0x00000000>;
+                assigned-address = <0x50>;
+        };
+        pmic5xxx_3_0: pmic@48,20400000000 {
+                reg = <0x48 0x204 0x00000000>;
+                assigned-address = <0x48>;
+        };
+        rcd_3_0: rcd@58,2C000000000 {
+                reg = <0x58 0x2C0 0x00000000>;
+                assigned-address = <0x58>;
+        };
+        //DIMM H
+        spd5118_3_1: spd@51,3C000000001 {
+                reg = <0x51 0x3C0 0x00000001>;
+                assigned-address = <0x51>;
+        };
+        pmic5xxx_3_1: pmic@49,20400000001 {
+                reg = <0x49 0x204 0x00000001>;
+                assigned-address = <0x49>;
+        };
+        rcd_3_1: rcd@59,2C000000001 {
+                reg = <0x59 0x2C0 0x00000001>;
+                assigned-address = <0x59>;
+        };
+        //DIMM I
+        spd5118_3_2: spd@52,3C000000002 {
+                reg = <0x52 0x3C0 0x00000002>;
+                assigned-address = <0x52>;
+        };
+        pmic5xxx_3_2: pmic@4a,20400000002 {
+                reg = <0x4a 0x204 0x00000002>;
+                assigned-address = <0x4a>;
+        };
+        rcd_3_2: rcd@5a,2C000000002 {
+                reg = <0x5a 0x2C0 0x00000002>;
+                assigned-address = <0x5a>;
+        };
+        //DIMM J
+        spd5118_3_3: spd@53,3C000000003 {
+                reg = <0x53 0x3C0 0x00000003>;
+                assigned-address = <0x53>;
+        };
+        pmic5xxx_3_3: pmic@4b,20400000003 {
+                reg = <0x4b 0x204 0x00000003>;
+                assigned-address = <0x4b>;
+        };
+        rcd_3_3: rcd@5b,2C000000003 {
+                reg = <0x5b 0x2C0 0x00000003>;
+                assigned-address = <0x5b>;
+        };
+        //DIMM K
+        spd5118_3_4: spd@54,3C000000004 {
+                reg = <0x54 0x3C0 0x00000004>;
+                assigned-address = <0x54>;
+        };
+        pmic5xxx_3_4: pmic@4c,20400000004 {
+                reg = <0x4c 0x204 0x00000004>;
+                assigned-address = <0x4c>;
+        };
+        rcd_3_4: rcd@5c,2C000000004 {
+                reg = <0x5c 0x2C0 0x00000004>;
+                assigned-address = <0x5c>;
+        };
+        //DIMM L
+        spd5118_3_5: spd@55,3C000000005 {
+                reg = <0x55 0x3C0 0x00000005>;
+                assigned-address = <0x55>;
+        };
+        pmic5xxx_3_5: pmic@4d,20400000005 {
+                reg = <0x4d 0x204 0x00000005>;
+                assigned-address = <0x4d>;
+        };
+        rcd_3_5: rcd@5d,2C000000005 {
+                reg = <0x5d 0x2C0 0x00000005>;
+                assigned-address = <0x5d>;
+        };
+};

 &i3c4 {
 	// P0 APML
--
2.25.1
